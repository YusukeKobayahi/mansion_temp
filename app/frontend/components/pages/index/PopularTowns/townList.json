{
  "towns": [
    {
      "name": "小石川",
      "id": "38876"
    },
    {
      "name": "本郷",
      "id": "38886"
    },
    {
      "name": "高輪",
      "id": "38766"
    },
    {
      "name": "白金",
      "id": "38762"
    },
    {
      "name": "南麻布",
      "id": "38775"
    },
    {
      "name": "南青山",
      "id": "38774"
    },
    {
      "name": "駒沢",
      "id": "39137"
    },
    {
      "name": "芝浦",
      "id": "38759"
    },
    {
      "name": "西麻布",
      "id": "38768"
    },
    {
      "name": "碑文谷",
      "id": "39044"
    },
    {
      "name": "神宮前",
      "id": "39187"
    },
    {
      "name": "晴海",
      "id": "38745"
    }
  ]
}
