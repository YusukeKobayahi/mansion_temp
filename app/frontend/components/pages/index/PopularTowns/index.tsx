import Link from "next/link";
import { buildLibraryPagePath } from "~/lib/utils";
import { event } from "~/lib/gtag";
import townList from "./townList.json";

import styles from "./index.module.scss";

export type TownListProps = {
  name: string;
  id: string;
};

const Prefectures: React.FC = () => {
  const filteredTownList: TownListProps[] = townList.towns;

  const clickTown = () => {
    event({
      action: "msl_area_click",
      category: "msl_top",
      label: "msl_town",
      value: 1,
    });
  };

  return (
    <section className={styles.container}>
      <h2 className={styles.title}>人気のエリアから探す</h2>
      <div className={styles.content}>
        <ul className={styles.list}>
          {filteredTownList.map((town) => {
            return (
              <li key={town.id} className={styles.item}>
                <Link
                  href={buildLibraryPagePath("towns", town.id)}
                  prefetch={false}
                  passHref
                >
                  <button className={styles.link} onClick={clickTown}>
                    {town.name}
                  </button>
                </Link>
              </li>
            );
          })}
        </ul>
      </div>
    </section>
  );
};

export default Prefectures;
