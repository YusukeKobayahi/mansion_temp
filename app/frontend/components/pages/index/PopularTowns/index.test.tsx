import { render } from "@testing-library/react";
import PopularTowns from "~/components/pages/index/PopularTowns";

describe("components/pages/index/PopularTowns", () => {
  it("正常に表示される", () => {
    const { asFragment } = render(<PopularTowns />);
    expect(asFragment()).toMatchSnapshot();
  });
});
