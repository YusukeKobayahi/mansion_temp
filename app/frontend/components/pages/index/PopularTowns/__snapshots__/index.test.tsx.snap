// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/index/PopularTowns 正常に表示される 1`] = `
<DocumentFragment>
  <section
    class="container"
  >
    <h2
      class="title"
    >
      人気のエリアから探す
    </h2>
    <div
      class="content"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38876"
          >
            小石川
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38886"
          >
            本郷
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38766"
          >
            高輪
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38762"
          >
            白金
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38775"
          >
            南麻布
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38774"
          >
            南青山
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/39137"
          >
            駒沢
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38759"
          >
            芝浦
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38768"
          >
            西麻布
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/39044"
          >
            碑文谷
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/39187"
          >
            神宮前
          </button>
        </li>
        <li
          class="item"
        >
          <button
            class="link"
            href="/towns/38745"
          >
            晴海
          </button>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
